### PR TITLE
Add basic Endpoints sample.

### PR DESCRIPTION
**Pull request by broady**
Updates #104.

### CODE REVIEW

File: appengine/endpoints/app.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+app.post('/echo', function(req, res) {
+  var msg = req.body.message;
+  console.log(req);
+  res.status(200).json({message: msg});
+});
+
+app.get('/auth/info/googlejwt', authInfoHandler);
+app.get('/auth/info/googleidtoken', authInfoHandler);
+
+function authInfoHandler(req, res) {
```
Comment (jonparrott) [inline appengine/endpoints/app.js]: I would actually move this above the app.get calls.

Comment (broady) [inline appengine/endpoints/app.js]: Done

File: appengine/endpoints/app.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+var express = require('express');
+var bodyParser = require('body-parser')
```
Comment (jonparrott) [inline appengine/endpoints/app.js]: missing ;. Run jshint over this?

**Comment by jonparrott** [comment]
This looks mostly good, with the exception of missing tests. We should allow @jmdobry to take a look before merging.

**Comment by jmdobry** [comment]
Build failing due to JSHint errors.

File: appengine/endpoints/app.js (hunk 2 above)
Comment (broady) [inline appengine/endpoints/app.js]: Done

**Comment by broady** [comment]
Fixed jshint errors.

**Comment by broady** [comment]
I'll squash this before we merge.

**Comment by broady** [comment]
I only found tests for `kraken` in the directory, so don't really have good examples of how to test this kind of app.

File: appengine/endpoints/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+```
+$ curl -vv -d '{"message":"foo"}' -H 'Content-Type: application/json' "${ENDPOINTS_HOST}/echo?key=${ENDPOINTS_KEY}"
+```
+
+If you're running locally, you won't need an API key.
+
+## Sending authenticated requests
+
+No Node.js client is written yet, but you can try the Python client found [here][python-client].
```
Comment (JustinBeckwith) [inline appengine/endpoints/README.md]: Do we have an issue tracking this somewhere?

File: appengine/endpoints/app.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+var express = require('express');
+var bodyParser = require('body-parser');
+
+var app = express();
+app.use(bodyParser.json());
+
+app.post('/echo', function(req, res) {
+  var msg = req.body.message;
+  console.log(req);
```
Comment (JustinBeckwith) [inline appengine/endpoints/app.js]: is this needed?

File: appengine/endpoints/app.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+runtime: nodejs
+vm: true
+
+skip_files:
```
Comment (JustinBeckwith) [inline appengine/endpoints/app.yaml]: this isn't needed anymore

File: appengine/endpoints/app.js (hunk 4 above)
Comment (broady) [inline appengine/endpoints/app.js]: Nope, fixed.

File: appengine/endpoints/README.md (hunk 3 above)
Comment (broady) [inline appengine/endpoints/README.md]: No. Will file issues for TODOs for this sample after this is submitted.

There are a few things:
* Client library
* Add tests
* Add error handling

File: appengine/endpoints/app.yaml (hunk 5 above)
Comment (broady) [inline appengine/endpoints/app.yaml]: It's still in the helloworld. Can you file a bug to fix it for all samples?

File: appengine/endpoints/README.md (hunk 3 above)
Comment (broady) [inline appengine/endpoints/README.md]: #107

File: appengine/endpoints/app.yaml (hunk 5 above)
Comment (broady) [inline appengine/endpoints/app.yaml]: #106

**Comment by JustinBeckwith** [comment]
LGTM

**Comment by jmdobry** [comment]
LGTM

**Pull request merged**